python-avd/pyavd/api/pool_manager/base_classes.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) 2023-2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import asdict, dataclass, field
+from typing import TYPE_CHECKING, Generic, TypeVar, cast
+
+from yaml import CSafeDumper, CSafeLoader, dump, load
+
+from pyavd.j2filters import natural_sort
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from typing_extensions import Self
+
+    from pyavd._eos_designs.shared_utils import SharedUtils
+
+T_ValueType = TypeVar("T_ValueType", int, str)
+T_PoolKeyType = TypeVar("T_PoolKeyType", bound="PoolKey")
+T_AssignmentKeyType = TypeVar("T_AssignmentKeyType", bound="AssignmentKey")
+
+FILE_HEADER = """\
+# This file is auto-generated by AVD eos_designs.
+# When there is a merge conflict for this file, rebase the branch, accept all current changes for this file, and re-run AVD.
+"""
+
+
+@dataclass(frozen=True)
+class PoolKey:
```

Comment:
some docstrings would be nice (othewise it is not obvious why we don't need just a Key class vs PoolKey and Assignment Key (yes we do, yes they are not the same))